### PR TITLE
Enables utils.compare_trees to help debugging

### PR DESCRIPTION
**Pull request by funkyfuture**
I find this actually helpful for developing an XML API's serialization routines.

----

dear diary, when i worked on this, the LLM that comes with my IDE for a while now actually inferred a useful auto-completion that excels prior methods for the first time. still underwhelming considering the kWh per completion proposal.

### CODE REVIEW

File: delb/utils.py [hunk 1]
```diff
@@ -1,30 +1,131 @@
...
+        assert isinstance(self.lhn, TagNode)
+        assert isinstance(self.rhn, TagNode)
+
+        if difference_kind is TreeDifferenceKind.TagAttributes:
+            return (
+                f"Attributes of tag nodes at {self.lhn.location_path} differ:\n"
+                f"{self.lhn.attributes}\n{self.rhn.attributes}"
+            )
+        elif difference_kind is TreeDifferenceKind.TagChildrenSize:
+            result = f"Child nodes of tag nodes at {self.lhn.location_path} differ:"
```
Comment (funkyfuture) [inline delb/utils.py:73]:
```suggestion
            result = f"Child nodes of tag node at {self.lhn.location_path} differ:"
```

File: delb/utils.py [hunk 2]
```diff
@@ -1,30 +1,131 @@
...
+            for a, b in zip_longest(
+                self.lhn.iterate_children(), self.rhn.iterate_children(), fillvalue=None
+            ):
+                result += f"\n\n{a!r}\n{b!r}"
+            return result
+        elif difference_kind is TreeDifferenceKind.TagLocalName:
+            return (
+                f"Local names of tag nodes at {self.lhn.location_path} differ: "
```
Comment (funkyfuture) [inline delb/utils.py:81]:
```suggestion
                f"Local names of tag node at {self.lhn.location_path} differ: "
```

File: delb/utils.py [hunk 3]
```diff
@@ -1,30 +1,131 @@
...
+                f"Nodes are of different type{parent_msg_tail} "
+                f"{self.lhn.__class__} != {self.rhn.__class__}"
+            )
+
+        assert isinstance(self.lhn, TagNode)
+        assert isinstance(self.rhn, TagNode)
+
+        if difference_kind is TreeDifferenceKind.TagAttributes:
+            return (
+                f"Attributes of tag nodes at {self.lhn.location_path} differ:\n"
```
Comment (funkyfuture) [inline delb/utils.py:69]:
```suggestion
                f"Attributes of tag node at {self.lhn.location_path} differ:\n"
```

**Comment by funkyfuture** [comment]
this now sits on top of #91.

**Review by JKatzwinkel — APPROVED** [review]

**Comment by funkyfuture** [comment]
for the record: the large chunks of uncovered (/-tested) code affects the string representation of encountered differences / errors. thus these code paths aren't reached when the other tests pass, but i've seen it all in action.

**Pull request merged**